src/classes/Indicators_Service.cls

```diff
@@ -1,67 +1,118 @@
+			enabled = (Boolean)field;
+		} else if(config.Type__c == 'Is Not Null') {
+			if(String.isNotBlank(String.valueOf(field)))   //String.isNotBlank((String)field)
+				enabled = true;
+		} else if(config.Type__c == 'Greater Than') {
+			if(field != null && String.isNotBlank(config.Contains_Value__c)) {
+				enabled = (Decimal.valueOf(String.valueOf(field)) > Decimal.valueOf(config.Contains_Value__c));
+			}
+		} else if(config.Type__c == 'Less Than') {
+			if(field != null && String.isNotBlank(config.Contains_Value__c)) {
+				enabled = (Decimal.valueOf(String.valueOf(field)) < Decimal.valueOf(config.Contains_Value__c));
+			}
+		} else if(config.Type__c == 'Blank') {
+			enabled = false;
+			display = false;
+		}
 
-    public class Indicator {
-        public string Color {get;set;}
-        public string Value {get;set;}
-        public Boolean Display {get;set;}
-        public string Hover {get;set;}
+		// include null option as TEXT type and first to avoid null pointer exceptions and accomodate Blank indicator  
+		if (config.Display_Value__c == null || (config.Display_Value__c.length() <= 4 && !spriteSet.contains(config.Display_Value__c))) {
+			iType = 'TEXT';
+		} else if (Pattern.matches('custom[0-9]{1,3}',config.Display_Value__c)) {
+			iType = 'CUSTOM_SPRITE';
+		} else if (spriteSet.contains(config.Display_Value__c)) {
+			iType = 'STANDARD_SPRITE'; 
+		} else {
+			iType = 'TEXT';
+			throw new WeathervaneException('No valid icon found for Display Value ' + config.Display_Value__c + '. Check Indicator record ' + config.DeveloperName);
+		}
+		
+		return new Indicator(config.Display_Value__c, enabled ? config.Enabled_Color__c : config.Disabled_Color__c, display, config.Hover_Text__c, iType);
+	}
 
-        public Indicator(String value, String color, Boolean display, String hover) {
-            this.Value = value; this.Color = color; this.Display = display; this.Hover = hover;
-        }
-    }
+	public static Set<String> spriteSet {
+		get {
+			if (spriteSet == null) {
+/*				Commenting out Org Cache of Set of sprite names for now
+				if (Cache.Org.contains('indicatorbar.partition1.spriteSet')) {
+					return (Set<String>)Cache.Org.get('indicatorbar.partition1.spriteSet');
+				} else {
+*/					
+					PageReference pr = new PageReference('/resource/sldswv/assets/icons/standard-sprite/svg/symbols.svg');
+					String content = '';
+					if (Test.isRunningTest()) {
+						content = Indicators_TEST.testSVG;
+					} else { 
+						content = pr.getContent().toString(); 
+					}
+					Pattern xmlid = Pattern.compile('id=\"[a-zA-Z_]+');
+					Matcher idMatcher = xmlid.matcher(content);
+					Set<String> idNames = new Set<String>();
+
+					if (idMatcher.find()) {
+					    do {
+					        idNames.add(idMatcher.group().substringAfter('="'));
+					    } while (idMatcher.find());
+					}
+					// see above re: org cache
+					// Cache.Org.put('indicatorbar.partition1.spriteSet',idNames);
+					return idNames;
```

Comment:
you'd need to assign the value to the spriteSet member for it to actually be memoized.